SlackAPI/SlackClientBase.cs

```diff
@@ -111,9 +130,18 @@ protected HttpWebRequest CreateWebRequest(Uri requestUri)
             return httpWebRequest;
         }
 
-        protected HttpResponseMessage PostRequest(string requestUri, MultipartFormDataContent form)
+        protected Task<HttpResponseMessage> PostRequestAsync(string requestUri, MultipartFormDataContent form, string token)
```

Comment:
FYI, this is the part I take issue with. We have a split client for handling async workflows specifically.

Calling `.Result` does not work as a way to invoke async from a sync call

Reply:
And I'm realizing we did do `.Result` before. Ugh.